File: Week 1/Problem Set 1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Prompt the user for change owed, in cents
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    // Initialize total coin count
    int coins = 0;

    // Calculate how many quarters you should give customer
    coins += calculate_quarters(cents);
    cents = cents % 25;

    // Calculate how many dimes you should give customer
    coins += calculate_dimes(cents);
    cents = cents % 10;

    // Calculate how many nickels you should give customer
    coins += calculate_nickels(cents);
    cents = cents % 5;

    // Calculate how many pennies you should give customer
    coins += calculate_pennies(cents);

    // Print the total number of coins
    printf("%d\n", coins);
}

// Calculate how many quarters to give
int calculate_quarters(int cents)
{
    return cents / 25;
}

// Calculate how many dimes to give
int calculate_dimes(int cents)
{
    return cents / 10;
}

// Calculate how many nickels to give
int calculate_nickels(int cents)
{
    return cents / 5;
}

// Calculate how many pennies to give
int calculate_pennies(int cents)
{
    return cents;
}

File: Week 6/Problem Set 6/sentimental-cash/cash.py
from cs50 import get_float


def main():
    # Prompt the user for change owed in dollars, convert to cents
    cents = -1
    while cents < 0:
        dollars = get_float("Change owed: ")
        cents = round(dollars * 100)

    # Initialize total coin count
    coins = 0

    # Calculate how many quarters you should give customer
    coins += calculate_quarters(cents)
    cents = cents % 25

    # Calculate how many dimes you should give customer
    coins += calculate_dimes(cents)
    cents = cents % 10

    # Calculate how many nickels you should give customer
    coins += calculate_nickels(cents)
    cents = cents % 5

    # Calculate how many pennies you should give customer
    coins += calculate_pennies(cents)

    # Print the total number of coins
    print(coins)

def calculate_quarters(cents):
    return cents // 25

def calculate_dimes(cents):
    return cents // 10

def calculate_nickels(cents):
    return cents // 5

def calculate_pennies(cents):
    return cents


if __name__ == "__main__":
    main()
